processors/conversion/convert_text.py

```diff
@@ -0,0 +1,230 @@
+"""
+Change the text in a dataset and write it to a new one
+"""
+import re
+import csv
+
+from common.lib.exceptions import ProcessorInterruptedException
+from backend.lib.processor import BasicProcessor
+from common.lib.helpers import UserInput
+
+__author__ = "Sal Hagen"
+__credits__ = ["Sal Hagen"]
+__maintainer__ = "Sal Hagen"
+__email__ = "4cat@oilab.eu"
+
+
+class ConvertText(BasicProcessor):
+    """
+    Retain only posts matching a given lexicon
+    """
+    type = "convert-text"  # job type ID
+    category = "Conversion"  # category
+    title = "Convert text"  # title displayed in UI
+    description = ("Changes text and outputs these in a new dataset. Converted text can also be added to the original "
+                   "dataset as annotations.")  # description displayed in UI
+    extension = "csv"
+
+    @classmethod
+    def get_options(cls, parent_dataset=None, config=None) -> dict:
+
+        options = {
+            "columns": {
+                "type": UserInput.OPTION_TEXT,
+                "default": "body",
+                "help": "Columns with texts to replace",
+            },
+            "find": {
+                "type": UserInput.OPTION_TEXT,
+                "default": "",
+                "help": "Text to replace",
+                "tooltip": "Multiple values can be replaced, separate with comma.",
+            },
+            "as_regex": {
+                "type": UserInput.OPTION_TOGGLE,
+                "default": False,
+                "help": "Interpret text to replace as regular expression",
+                "tooltip": "Regular expressions are parsed with Python",
+            },
+            "replace": {
+                "type": UserInput.OPTION_TEXT,
+                "default": "",
+                "help": "Text to insert",
+                "tooltip": "This can only be a single string value. Whitespace characters will also be inserted.",
+            },
+            "whitespace-notice": {
+                "type": UserInput.OPTION_INFO,
+                "help": "Note: whitespace characters are taken into account. Using 'cat, dog' as text to replace will "
+                "match all occurrences of 'cat' but only ' dog', starting with a whitespace. This also goes "
+                "for replacement text: replacing 'cat' with ' cat ' will add whitespaces.",
+            },
+            "case-sensitive": {
+                "type": UserInput.OPTION_TOGGLE,
+                "default": False,
+                "help": "Case sensitive",
+            },
+            "only-keep-replacements": {
```

Comment:
This makes the processor both a filter and a converter,  but I think I would prefer it without this option, and then you can use Filter by Value to remove fields that don't contain the relevant text. Means we don't have to implement that functionality twice and makes it clearer at a glance what the processor can and cannot do...